client/utils.js

```diff
@@ -7,3 +7,8 @@ export function pathToRegExp( path ) {
 	}
 	return new RegExp( '^' + path + '(/.*)?$' );
 }
+
+export function isFullSiteEditingTheme( theme ) {
+	const features = theme?.taxonomies?.theme_feature;
+	return features && features.some( ( feature ) => feature.slug === 'block-templates' );
+}
```

Comment:
I think we might also need to check beta eligibility/status before adding the badges?

That is, soon we will roll out Universal themes to all users allowing non-beta users to activate them and use as the standard/classic experience.  We wouldn't want the beta badge showing up on these Universal themes in that situation. 🤔

Reply:
I'm curious how you feel about using the `wpcom_is_fse_activated` option to disable / enable the beta badge instead.

I added a guard to evaluate `isSiteUsingCoreSiteEditor` selector, which references the `is_core_site_editor_enabled` attribute. The attribute is fetched from the rest API which is resolved from the `wpcom_is_fse_activated` option. 

The beta badge should only be displayed on a core FSE enabled site. One thing to note, however, is that using the `wpcom_is_fse_activated` option means that if a user enrolled in the beta disables the Appearance > Site Editor "Enable Site Editor" setting, all beta badges will disappear on block-based themes.

This makes sense to me because the user would then be opted into the standard/classic experience of Universal Themes, but I'm curious what your thoughts are.

Reply:
Im not entirely certain either way.

>  if a user enrolled in the beta disables the Appearance > Site Editor "Enable Site Editor" setting, all beta badges will disappear on Universal / block-based themes.

At this point, that toggle isnt available to users.  And whether or not we add a toggle for users to do this sort of depends on the p2 discussion @creativecoder brought up.

Whether or not it makes sense for that to remove the beta tag also depends on where/how we end up having this sort of toggle.  If it is a prompt on theme switch, then those themes probably still need the beta tag to convey to the user that they can get back to the beta experience by activating them.  If that toggle is elsewhere, like at the top of the showcase itself, or in the 'beta testing' menu item, then it seems like it would be ok to hide the 'beta' tags on the theme cards when that toggle is off. 🤔 

TLDR - I think it depends on what/how we set up that toggle and present it to the user.

Reply:
> At this point, that toggle isnt available to users. And whether or not we add a toggle for users to do this sort of depends on the p2 discussion @creativecoder brought up recently.
>
> Whether or not it makes sense for that to remove the beta tag also depends on where/how we end up having this sort of toggle.

Gotcha -- I mistakenly thought it would be made available to users upon release of the beta. Thanks for the info Addie 👍

Reply:
Great question to ask! I think tying the Beta badge to the `core-site-editor-enabled` blog sticker makes the most sense to me.

The main reason being that once the beta is over, we'll likely keep the `wpcom_is_fse_activated` option and related functionality (but probably default it to on) so that users can still have a "classic" theme experience, if they want. But I imagine we'll remove all of the visual indications of "Beta", and ignore the `core-site-editor-enabled` sticker at the same time, once that's considered the default functionality for new sites.